testsuite/MDAnalysisTests/core/test_atomselections.py

```diff
@@ -1218,3 +1231,112 @@ def test_record_type_sel():
+    assert len(u.select_atoms("chainID A")) == len(u.atoms)
+
+
+@pytest.fixture()
+def u_fake_masses():
+    u = mda.Universe(TPR)
+    u.atoms[-10:].masses = - (np.arange(10) + 1.001)
+    u.atoms[:5].masses = 0.1 * 3  # 0.30000000000000004
+    u.atoms[5:10].masses = 0.30000000000000001
+    return u
+
+
+@pytest.mark.parametrize("selstr,n_atoms, selkwargs", [
+    ("mass 0.8 to 1.2", 23844, {}),
+    ("mass 8e-1 to 1200e-3", 23844, {}),
+    ("mass -5--3", 2, {}),  # select -5 to -3
+    ("mass -5 --3", 2, {}),  # spacing
+    ("mass -5- -3", 2, {}),  # spacing
+    ("mass -3 : -5", 0, {}),  # wrong way around
+    # float equality
+    ("mass 0.3", 5, {"rtol": 0, "atol": 0}),
+    ("mass 0.3", 5, {"rtol": 1e-22, "atol": 1e-22}),
+    # 0.30000000000000001 == 0.3
+    ("mass 0.3 - 0.30000000000000004", 10, {}),
+    ("mass 0.30000000000000004", 5, {"rtol": 0, "atol": 0}),
+    ("mass 0.3 0.30000000000000001", 5, {"rtol": 0, "atol": 0}),
+    # float near-equality
+    ("mass 0.3", 10, {}),
+    ("mass 0.30000000000000004", 10, {}),
+    ("mass 0.3 0.30000000000000001", 10, {}),
+    # prop thingy
+    ("prop mass == 0.3", 10, {}),
+    ("prop mass == 0.30000000000000004", 10, {}),
+    ("prop mass == 0.30000000000000004", 5, {"rtol": 0, "atol": 0}),
+])
+def test_mass_sel(u_fake_masses, selstr, n_atoms, selkwargs):
+    # test auto-topattr addition of float (FloatRangeSelection)
+    ag = u_fake_masses.select_atoms(selstr, **selkwargs)
+    assert len(ag) == n_atoms
+
+def test_mass_sel_warning(u_fake_masses):
+    warn_msg = (r"Using float equality .* is not recommended .* "
+                r"we recommend using a range .*"
+                r"'mass -0.6 to 1.4'.*"
+                r"use the `atol` and `rtol` keywords")
+    with pytest.warns(SelectionWarning, match=warn_msg):
+        u_fake_masses.select_atoms("mass 0.4")
+
+
+@pytest.mark.parametrize("selstr,n_res", [
+    ("resnum -10 to 3", 14),
+    ("resnum -5--3", 3),  # select -5 to -3
+    ("resnum -3 : -5", 0),  # wrong way around
+])
+def test_int_sel(selstr, n_res):
+    # test auto-topattr addition of int (IntRangeSelection)
+    u = mda.Universe(TPR)
+    u.residues[-10:].resnums = - (np.arange(10) + 1)
+    ag = u.select_atoms(selstr).residues
+    assert len(ag) == n_res
+
+
+def test_negative_resid():
+    # this is its own separate test because ResidSelection
+    # has special matching for icodes
+    text = """\
+    ATOM      1  N   ASP A  -1      19.426  19.251   2.191  1.00 59.85           N
+    ATOM      2  CA  ASP A  -1      20.185  18.441   1.255  1.00 54.54           C
+    ATOM      3  C   ASP A  -1      21.660  18.901   1.297  1.00 40.12           C
+    ATOM      4  O   ASP A  -1      21.958  19.978   1.829  1.00 35.85           O"""
+    u = mda.Universe(StringIO(textwrap.dedent(text)), format="PDB")
+    ag = u.select_atoms("resid -1")
+    assert len(ag) == 4
+
+
+@pytest.mark.parametrize("selstr, n_atoms", [
+    ("aromaticity", 5),
```

Comment:
So I think `aromaticity` is the wrong word for this, should be `not aromatic`.  Should there be some way of overriding the default keyword used by the TopologyAttrMeta or should exceptions just require a Selection class to be manually written (current behaviour)

Reply:
``aromatic`` still exists, the metaclass just dumbly checks whether the attrname (aromaticity in this case) exists in `_SELECTIONDICT`; possibly if we switched the contents of _SELECTIONDICT to be classes instead, we could avoid the redundancy? As it is we could just not document weird quirks like this?

To me the options are:
* change the attribute name to "aromatic"
* change _SELECTIONDICT to have classes instead
* do nothing, hope people don't notice the double up